src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -65,7 +66,11 @@ template Batch
             (instructionCid, instruction) <- fetchByKey @Instruction (requestors, instructionId)
             exercise instructionCid Instruction.Execute
         l <- mapA settleInstruction instructionIds
-        -- undisclose accounts from settler
+        -- check consistency
+        isConsistent <- (== 1) . length . dedup <$> mapA (\holdingCid ->
+            exercise (coerceContractId holdingCid : ContractId Base.I) Base.GetTemplateTypeRepText with actor = settler
+          ) l
+        assertMsg "The template type rep of allocated holdings must be the same for all instructions." isConsistent
```

Comment:
Could the same check be performed with the (non-serialisable) type-rep retrieved from the interface? It would spare us from defining an additional choice

Reply:
The `settler` does not have the authorization to `fetch`, so it is not a one line change., but can store the template type rep text (in the Instruction) at the time of allocation. That way we can get rid of the `GetTemplateTypeRepText` choice of the `D.F.I.Holding.Base`.

Reply:
Right.. 

I don't have a strong opinion on which approach is preferable:
- the first one exposes `typeRep` via a choice in the holding (that a customer potentially needs to implement)
- the second stores the `typeRep` on the ledger with the instruction, but a customer doesn't need to do much

Any preference from your side?

Reply:
I think the latter is better, that is what I have in the last commit.